http-server/src/main/java/org/triplea/server/error/reporting/ErrorReportController.java

```diff
@@ -9,38 +9,37 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.ws.rs.POST;
 import javax.ws.rs.Path;
-import javax.ws.rs.Produces;
 import javax.ws.rs.core.Context;
-import javax.ws.rs.core.MediaType;
 import lombok.Builder;
-import org.triplea.http.client.error.report.ErrorUploadClient;
-import org.triplea.http.client.error.report.ErrorUploadRequest;
-import org.triplea.http.client.error.report.ErrorUploadResponse;
+import org.triplea.http.client.error.report.ErrorReportClient;
+import org.triplea.http.client.error.report.ErrorReportRequest;
+import org.triplea.http.client.error.report.ErrorReportResponse;
+import org.triplea.server.http.HttpController;
 import org.triplea.server.http.IpAddressExtractor;
 
 /** Http controller that binds the error upload endpoint with the error report upload handler. */
 @Builder
-@Produces(MediaType.APPLICATION_JSON)
-@Path("/")
-public class ErrorReportController {
-  @Nonnull private final Function<ErrorReportRequest, ErrorUploadResponse> errorReportIngestion;
+public class ErrorReportController extends HttpController {
+  @Nonnull
+  private final Function<org.triplea.server.error.reporting.ErrorReportRequest, ErrorReportResponse>
+      errorReportIngestion;
 
   @POST
-  @Path(ErrorUploadClient.ERROR_REPORT_PATH)
+  @Path(ErrorReportClient.ERROR_REPORT_PATH)
   @RateLimited(
       keys = {KeyPart.IP},
       rates = {
-        @Rate(limit = ErrorUploadClient.MAX_REPORTS_PER_DAY, duration = 1, timeUnit = TimeUnit.DAYS)
+        @Rate(limit = ErrorReportClient.MAX_REPORTS_PER_DAY, duration = 1, timeUnit = TimeUnit.DAYS)
       })
-  public ErrorUploadResponse uploadErrorReport(
-      @Context final HttpServletRequest request, final ErrorUploadRequest errorReport) {
+  public ErrorReportResponse uploadErrorReport(
+      @Context final HttpServletRequest request, final ErrorReportRequest errorReport) {
 
     if (errorReport.getBody() == null || errorReport.getTitle() == null) {
       throw new IllegalArgumentException("Missing error report body and/or title");
     }
 
     return errorReportIngestion.apply(
-        ErrorReportRequest.builder()
+        org.triplea.server.error.reporting.ErrorReportRequest.builder()
```

Comment:
Why are there 2 `ErrorReportRequest` classes?

Reply:
Name collision, not always the end of the world, but the handling is so close to one another that I agree it's confusing. I resolved this by splitting up the second ErrorReportRequest, it was a value object holding IP and the original error report. It's now split up and the second value object deleted.